java/src/main/java/com/nedap/retail/example/Application.java

```diff
@@ -198,45 +199,33 @@ private static void retrieveGroupInfo(final GroupInfo.Group group) {
+    private static void retrieveGroupInfo() throws MessageParsingException, HttpRequestException {
         final GroupInfo groupInfo = api.retrieveGroupInfo();
-        if (groupInfo != null && groupInfo.groups != null) {
-            LOG.info("Group information");
-            for (final GroupInfo.Group group : groupInfo.groups) {
-                retrieveGroupInfo(group);
-            }
-        } else {
-            throw new MessageParsingException("Group info could not be retrieved");
+        LOG.info("Group information");
+        for (final GroupInfo.Group group : groupInfo.groups) {
+            logGroupInfo(group);
         }
     }
 
-    private static void retrieveSystemStatus() throws Exception {
+    private static void retrieveSystemStatus() throws MessageParsingException, HttpRequestException {
         final SystemStatus status = api.retrieveSystemStatus();
-        if (status != null) {
-            LOG.info("System status");
-            LOG.info("Unreachable units: {}", status.unreachableUnits != 0);
-            LOG.info("Device management connection error: {}", status.deviceManagementConnectionError != 0);
-            LOG.info("Rfid reader errors: {}", status.rfidErrors != 0);
-            LOG.info("Blocked IR beam sensors: {}", status.blockedIrBeamSensors != 0);
-        } else {
-            throw new MessageParsingException("System status could not be retrieved");
-        }
+        LOG.info("System status");
+        LOG.info("Unreachable units: {}", status.unreachableUnits != 0);
+        LOG.info("Device management connection error: {}", status.deviceManagementConnectionError != 0);
+        LOG.info("Rfid reader errors: {}", status.rfidErrors != 0);
+        LOG.info("Blocked IR beam sensors: {}", status.blockedIrBeamSensors != 0);
     }
 
-    private static void retrieveSystemSettings() throws Exception {
+    private static void retrieveSystemSettings() throws MessageParsingException, HttpRequestException {
         final Settings settings = api.retrieveSystemSettings();
-        if (settings != null) {
-            LOG.info("System settings");
-            LOG.info("RF enabled {}", settings.enableRf);
-            LOG.info("RFID enabled {}", settings.enableRfid);
-            LOG.info("RF alarm triggers {}", settings.lightSoundRf);
-            LOG.info("RFID alarm triggers {}", settings.lightSoundRfid);
-        } else {
-            throw new MessageParsingException("System settings could not be retrieved");
-        }
+        LOG.info("System settings");
+        LOG.info("RF enabled {}", settings.enableRf);
+        LOG.info("RFID enabled {}", settings.enableRfid);
+        LOG.info("RF alarm triggers {}", settings.lightSoundRf);
+        LOG.info("RFID alarm triggers {}", settings.lightSoundRfid);
     }
 
-    private static void updateSystemSettings(final BufferedReader inputBuffer) throws Exception {
+    private static void updateSystemSettings(final BufferedReader inputBuffer) throws IOException {
```

Comment:
Because `readBoolean` throws an `IOException` which in turn gets it from `BufferedReader`. I could add some abstraction to rethrow `IOExceptions` on user input. Maybe as `MessageParsingException`.